File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
       
        int i;
        for(i=0;i<arr.length;i++){
            if(arr[i]==0){
                break;
            }
        }
        System.out.println(i);
        
        for(int j=i+1;j<arr.length;j++){
            if(arr[j]!=0){
                int temp=arr[j];
                arr[j]=arr[i];
                arr[i]=temp;
                i++;
            }
        }
        
    }
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int left=0;
        while(nums[left]!=0&&left<nums.size()){
            left++;
        }
        for(int i=left+1;i<nums.size();i++){
            if(nums[i]!=0){
                swap(nums[left],nums[i]);
                left++;
            }
        }
        
    }
};